app/models/track.rb

```diff
@@ -1,6 +1,6 @@
 class Track < ApplicationRecord
   has_many :users
-  has_many :track_courses
+  has_many :track_courses, dependent: :destroy
```

Comment:
What do you think about using `dependent: :delete_all` instead?

I don't think we have any callbacks on `TrackCourse`. So there's no need to instantiate objects and delete them one by one. We can delete them all with a single `DELETE` SQL query.